.circleci/config/jobs/proto-and-api-deltas.yml

```
@@ -0,0 +1,18 @@
+executor: go-machine-medium
+steps:
+- checkout
+- install-go
+- run:
+    name: "Check proto deltas"
+    command: |
+      make proto
+      rm -f diffFile
+      git diff > diffFile
+      [ -s diffFile ] && exit -1
```

You can eliminate the file here by leveraging an option to git diff:

```
git diff --exit-code
```

From `git diff --help`:
```
--exit-code
    Make the program exit with codes similar to diff(1). That is, it exits with 1 if there were
    differences and 0 means no differences.
```

This also has a nice benefit of producing the full diff output, so if this does fail the CI output will provide something to compare against locally.

However, this will still return 0 in the case where untracked files are added. So for example, if a new protobuf file was added that would result in a new generated file, but the generated file was not added. To detect this, in the past I have used `git status --porcelain` which will have no output if there are no changes or untracked files. So a command like:

```
test -z $(git status --porcelain)
```
That will return a non-0 if the git status command has any output.

So to summarize all of that I would suggest:

```suggestion
      git diff --exit-code
      git status --porcelain
      test -z $(git status --porcelain)
```